### PR TITLE
Create best-practices.mdx

### PR DESCRIPTION
**Pull request by tilucasoli**
## Describe your changes
* Create a documentation for best practices
* Create `StyleRecipe` as a `abstract class`
* Change all the Durations inside Pressable and PressableBox to 0, it's importante because we couldn't change the current behavior.

## Type of change
- **Docs** (changes to the documentation)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: website/pages/docs/guides/best-practices.mdx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+## Defining Variants
+To create variants is quite simple. You just need to create a class that extends `Variant` and define the variants you want to use, so I'll create a class called `ShadcnButtonVariants` and define the cases `primary`, `destructive` and `link`. Shadcn has more than that, but for this example, I'll use only those three.
+
+```dart
+class ButtonType extends Variant {
+  const ButtonType._(super.name);
+
+  static const primary = ButtonType._('primaryButtonVariant');
+  static const destructive = ButtonType._('destructiveButtonVariant');
+  static const link = ButtonType._('linkButtonVariant');
```
[the remark this answers is not in the document]
Comment (tilucasoli) [reply website/pages/docs/guides/best-practices.mdx]: Done

File: website/pages/docs/guides/best-practices.mdx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    this.type = ButtonType.primary,
+  });
+
+  final String title;
+  final ButtonType type;
+
+  @override
+  Widget build(BuildContext context) {
+
+    return Pressable(
```
[the remark this answers is not in the document]
Comment (tilucasoli) [reply website/pages/docs/guides/best-practices.mdx]: I renamed `GestureBox` to `PressureBox` as we spoke

File: website/pages/docs/guides/best-practices.mdx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    ),
+    ButtonType.destructive(
+      backgroundColor.redAccent(),
+    ),
+    ButtonType.link(
+      backgroundColor.transparent(),
+    ),
+  ).applyVariant(type);
+
+  late final title = Style(
```
[the remark this answers is not in the document]
Comment (tilucasoli) [reply website/pages/docs/guides/best-practices.mdx]: As it is a simple case, I didn't see why to use `StyleRecipe`, but I already renamed it and brought it again to Mix package.

**Review by leoafarias — CHANGES_REQUESTED** [review]

File: lib/src/utils/style_recipe.dart [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import '../../mix.dart';
```
[the remark this answers is not in the document]
Comment (tilucasoli) [reply lib/src/utils/style_recipe.dart]: Done

**Review by leoafarias — APPROVED** [review]
LGTM

**Pull request merged**